core/types/transaction.go

```diff
@@ -245,9 +267,19 @@ func (tx *Transaction) WithSignature(signer Signer, sig []byte) (*Transaction, e
 	return cpy, nil
 }
 
+func (tx *Transaction) WithAASignature() *Transaction {
+	cpy := &Transaction{data: tx.data}
+	cpy.data.V = big.NewInt(27)
```

Comment:
Can we update this so that it respects the EIP-155 chain id that already existed in v?

Reply:
So this method is only supposed to be called on newly created transactions that have not yet been signed, analogous to `tx.WithSignature()`. `v` should not already be set, otherwise there would be no need for this to be called. In general, this is just a helper function for the AA tests and is not used during normal geth operation.